module/userProjects/helper.js

```diff
@@ -159,98 +212,98 @@ module.exports = class UserProjectsHelper {
-                //         // create new program using current name and add existing solution and remove program from it.
-                //     }
-                // }
+                    if (!userProject[0].solutionInformation) {
+                        createNewProgramAndSolution = true;
+                    } else {
+                        solutionExists = true;
+                        // create new program using current name and add existing solution and remove program from it.
+                    }
+                }
 
-                // let addOrUpdateEntityToProject = false;
+                let addOrUpdateEntityToProject = false;
 
-                // if (data.entityId) {
+                if (data.entityId) {
                     
-                //     // If entity is not present in project or new entity is updated.
-                //     if (
-                //         !userProject[0].entityInformation ||
-                //         (
-                //             userProject[0].entityInformation &&
-                //             userProject[0].entityInformation._id !== data.entityId
-                //         )
-                //     ) {
-                //         addOrUpdateEntityToProject = true;
-                //     }
-                // }
+                    // If entity is not present in project or new entity is updated.
+                    if (
+                        !userProject[0].entityInformation ||
+                        (
+                            userProject[0].entityInformation &&
+                            userProject[0].entityInformation._id !== data.entityId
+                        )
+                    ) {
+                        addOrUpdateEntityToProject = true;
+                    }
+                }
                 
-                // if (addOrUpdateEntityToProject) {
+                if (addOrUpdateEntityToProject) {
 
-                //     let entityInformation =
-                //         await _entitiesInformation([data.entityId]);
+                    let entityInformation =
+                        await _entitiesInformation([data.entityId]);
 
-                //     if (!entityInformation.success) {
-                //         return resolve(entityInformation);
-                //     }
+                    if (!entityInformation.success) {
+                        return resolve(entityInformation);
+                    }
 
-                //     updateProject["entityInformation"] = entityInformation.data[0];
-                //     updateProject.entityId = entityInformation.data[0]._id;
-                // }
+                    updateProject["entityInformation"] = entityInformation.data[0];
+                    updateProject.entityId = entityInformation.data[0]._id;
+                }
                 
-                // if (createNewProgramAndSolution || solutionExists) {
+                if (createNewProgramAndSolution || solutionExists) {
```

Comment:
@Prajwal17Tunerlabs createNewProgramAndSolution this you will get undefined here if you are not defined above